operator/hack/common.sh

```diff
@@ -34,26 +35,27 @@ function apply_operator_manifests() {
   catalog_source_crd="$("${ROOT_DIR}/operator/hack/retry-kubectl.sh" < /dev/null get customresourcedefinitions.apiextensions.k8s.io catalogsources.operators.coreos.com -o yaml)"
   local disable_security_context_config="# "
   local has_scc_key
+  local operator_channel
+  local install_plan_approval
+  local starting_csv
   has_scc_key="$(yq eval '[.. | select(key == "grpcPodConfig")].[].properties | has("securityContextConfig")' - <<< "$catalog_source_crd")"
   if [[ "$has_scc_key" == "true" ]]; then
       disable_security_context_config=""
   fi
 
   if [[ "${USE_MIDSTREAM_IMAGES}" == "true" ]]; then
-    # Get Operator channel from json for midstream
-    operator_channel=$(< midstream/iib.json jq -r '.operator.channel')
-  env -i PATH="${PATH}" \
-    INDEX_VERSION="${index_version}" OPERATOR_VERSION="${operator_version}" NAMESPACE="${operator_ns}" OPERATOR_CHANNEL="${operator_channel}" \
-    IMAGE_TAG_BASE="${image_tag_base}" DISABLE_SECURITY_CONTEXT_CONFIG="${disable_security_context_config}" \
-    envsubst < "${ROOT_DIR}/operator/hack/operator-midstream.envsubst.yaml" \
-    | "${ROOT_DIR}/operator/hack/retry-kubectl.sh" -n "${operator_ns}" apply -f -
+    install_plan_approval="Automatic"
+    starting_csv="null"
```

Comment:
So strictly speaking, users of midstream images won't get `csv_version` they desire even if they provide something on the command line, right?

I think it's better to fail with error in case `install_plan_approval == Automatic && csv_version != index_image_tag` because users won't get what they asked for.

Reply:
> So strictly speaking, users of midstream images won't get csv_version they desire even if they provide something on the command line, right?

Right, but that is also the case now. For example on current `master` [this setting](https://github.com/stackrox/stackrox/blob/db45ed40fd0f4d82e32c2c647380b93bd0637a6a/tests/e2e/lib.sh#L189) and following `export` has no impact. The `OPERATOR_VERSION` value does get plumbed by the `Makefile` through to `olm-operator-install.sh` and the `envsubst` call, but the old midstream template does not even use this variable. So the `operator.version` field in `iib.json` is effectively unused - the latest version in the `stable` channel gets installed.

> I think it's better to fail with error

Perhaps, but this would:
- be a change in semantics,
- require syncing with the team responsible for Power/Z (w.r.t. what is really intended)
- arguably be outside the scope of this PR
- slow down the konflux effort

So I'd prefer not to change this here.

Reply:
Since the change already touches P/Z, I think, we need to do some cleanups to make things less confusing. I'm sure you spent some time figuring things out and I also spent time trying to understand how things work as part of this review. There's redundant stuff that must go away to not take away time from us and colleagues in the future.

Let me try suggest something that does not change semantics that much.